### ISSUE TITLE
How to request with cookie  in dio 4.0.0

### ISSUE THREAD

**Issue by jerryzhoujw** [cfug/dio#1298]
### New Issue Checklist

- [x ] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |   flutter   |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |  2.2.3    |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |    4.0.0  |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |  vscode    |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

The original solution with cookie is from as bellow
[https://stackoverflow.com/questions/64497445/how-to-pass-cookie-session-id-in-dio-resquest](https://stackoverflow.com/a/66915057/1530581)

```dart
dio.interceptors.add(CookieManager(cookieJar));
```

in dio 4.0.0

I didn't found way to create `CookieManager` with `cookieJar`, how to request with cookie ?
